js/src/setup-ads/ads-stepper/index.js

```diff
@@ -27,11 +30,45 @@ import {
  */
 const AdsStepper = ( { formProps } ) => {
 	const [ step, setStep ] = useState( '1' );
+	const initHasAdsConnectionRef = useRef( null );
+	const { hasFinishedResolution: hasResolvedGoogleAccount } =
+		useGoogleAccount();
+
+	const {
+		googleAdsAccount,
+		hasFinishedResolution: hasResolvedGoogleAdsAccount,
+		hasGoogleAdsConnection,
+	} = useGoogleAdsAccount();
+
+	const {
+		hasAccess,
+		hasFinishedResolution: hasResolvedAdsAccountStatus,
+		step: adsAccountSetupStep,
+	} = useGoogleAdsAccountStatus();
+
 	useEventPropertiesFilter( FILTER_ONBOARDING, {
 		context: CONTEXT_ADS_ONBOARDING,
 		step,
 	} );
 
+	if (
+		! hasResolvedGoogleAccount ||
+		! hasResolvedGoogleAdsAccount ||
+		! hasResolvedAdsAccountStatus ||
+		googleAdsAccount === null
+	) {
+		return null;
+	}
```

Comment:
Before this PR, it only flashes the loading spinner within the Google Ads Card:

https://github.com/user-attachments/assets/1b91bacb-bfb5-4ebd-aac1-dd06f57eacc9

Checking `hasFinishedResolution` states here leads to almost the entire page flashing momentarily during the account connection.

https://github.com/user-attachments/assets/de8d2727-e0db-464d-8d8b-420438ffe7b8

Could it be kept as close to the original as possible?